lib/fluent/plugin/out_cloudwatch_logs.rb

```diff
@@ -124,8 +138,31 @@ def write(chunk)
         end
 
         unless log_group_exists?(group_name)
+          #rs = [[name, timestamp, record],[name,timestamp,record]]
+          #get tags and retention from first record
+          #as we create log group only once, values from first record will persist
+          record = rs[0][2]
+
+          awstags = @log_group_aws_tags
+          unless @log_group_aws_tags_key.nil?
+            if @remove_log_group_aws_tags_key
+              awstags = record.delete(@log_group_aws_tags_key)
+            else
+              awstags = record[@log_group_aws_tags_key]
+            end
+          end
+
+          retention_in_days = @retention_in_days
+          unless @retention_in_days_key.nil?
+            if @remove_retention_in_days_key
+              awstags = record.delete(@retention_in_days_key)
```

Comment:
s/awstags/retention_in_days/ ?

Reply:
thanks a lot for the review and catching this issue. Fixed